virttest/qemu_devices/qcontainer.py

```diff
@@ -2395,9 +2399,13 @@ def define_hbas(qtype, atype, bus, unit, port, qbus, pci_bus, iothread,
             devices[-1].set_param('port', unit)
             devices[-1].set_param('removable', removable, bool)
         elif fmt == 'floppy':
-            # Overwrite qdevices.QDevice with qdevices.QFloppy
-            devices[-1] = qdevices.QFloppy(unit, 'drive_%s' % name, name,
-                                           ({'busid': 'drive_%s' % name}, {'type': fmt}))
+            if Flags.FLOPPY_DEVICE in self.caps:
+                devices[-1] = qdevices.QDevice('floppy',
+                                               child_bus=qdevices.QPCIBus('floppy', 'PCI'))
```

Comment:
Suggest the following change to respect the workflow of the `images_define_by_variables` procedure.
```suggestion
                devices[-1] = qdevices.QDevice('floppy', aobject=name)
                devices[-1].parent_bus += ({'busid': 'drive_%s' % name},)
                devices[-1].set_param('unit', unit)
```